### PR TITLE
Support to import third party .deb packages using reprepro-updater

### PR DESCRIPTION
**Pull request by j-rivero**
The PR implements the generation of a job called "reprepro-importer" that supports the injection of new packages in our repositories. The PR contains:

 * DSL generation of the new job
 * The script to be run in the job,  basically a fork (the repo is only accepting PR for critical fixes) of `reprepro-updater`. My changes can be checked [here](https://github.com/ros-infrastructure/reprepro-updater/compare/refactor...j-rivero:refactor), mostly implements fail-fast, custom repository values and configuration to import Dart packages from our PPA.

Waiting some changes in chef-osrf to provision a testing repository in packages.osrfoundation.org to do the final testing.

Final testing using `ubuntu_testing` repository in the packages machine, some scenarios:

 * Do not override current distributions file if the one generated by the reprepro-updater tool is different. [Show that diff](https://build.osrfoundation.org/job/reprepro_importer/24/console#console-section-1) in console.
 * Simulation mode with no real package uploading but [listing packages to upload](https://build.osrfoundation.org/job/reprepro_importer/26/console#console-section-1)
 * Real [upload of packages ](https://build.osrfoundation.org/job/reprepro_importer/28/console)

**TODO:**
The PR needs support for verify_release [in the configuration file](https://github.com/ros-infrastructure/reprepro-updater/blob/refactor_osrfbuild/config/packages.osrfoundation.org/openrobotics_dart_packages_from_ppa.yaml#L9). @nuclearsandwich commented that is not as simple as adding the key but the key needs to be imported first to the system.

### CODE REVIEW

File: jenkins-scripts/docker/reprepro_updater.bash [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -x
+
+set -e
+
+# Knowing Script dir beware of symlink
+[[ -L "${0}" ]] && SCRIPT_DIR=$(readlink "${0}") || SCRIPT_DIR="${0}"
+SCRIPT_DIR="${SCRIPT_DIR%/*}"
+
+PYTHON_VENV="${WORKSPACE}/venv"
+REPREPRO_GIT_REPO="https://github.com/j-rivero/reprepro-updater"
```
Comment (nuclearsandwich) [inline jenkins-scripts/docker/reprepro_updater.bash:10]: Should the bug be fixed upstream so the fork isn't needed? I don't see a PR associated with this issue?

File: jenkins-scripts/docker/reprepro_updater.bash [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+REPREPRO_REPO_PATH="$WORKSPACE/repo"
+
+REPREPRO_PARAMS=''
+if ${COMMIT}; then
+    REPREPRO_PARAMS='--commit'
+fi
+
+if [[ ! -f "${HOME}/.buildfarm/reprepro-updater.ini" ]]; then
+    echo "The reprepro configuration file was not found"
+    exit -1
```
Comment (nuclearsandwich) [inline jenkins-scripts/docker/reprepro_updater.bash:21]: Does -1 / 255 have any specific significance as an exit code?

File: jenkins-scripts/docker/reprepro_updater.bash [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+rm -fr ${REPREPRO_REPO_PATH}
+git clone "${REPREPRO_GIT_REPO}" -b "${REPREPRO_GIT_BRANCH}" "${REPREPRO_REPO_PATH}"
+echo '# END SECTION'
+
+export PYTHONPATH="${REPREPRO_REPO_PATH}/src"
+export GNUPGHOME=/var/lib/jenkins/.gnupg 
+
+echo '# BEGIN SECTION: run reprepro'
+cd "${REPREPRO_REPO_PATH}/scripts"
+sudo -E bash -c "PYTHONPATH=${REPREPRO_REPO_PATH}/src python3 import_upstream.py ${REPREPRO_PARAMS} \
```
Comment (nuclearsandwich) [inline jenkins-scripts/docker/reprepro_updater.bash:47]: Why is reprepro-updater being run as root here?

File: jenkins-scripts/docker/reprepro_updater.bash (hunk 1 above)
Comment (j-rivero) [reply jenkins-scripts/docker/reprepro_updater.bash:10]: Problem is that I think changes are not strictly bugs but changes on behavior. With the option of only accepting critical fixes in that repo I don't think that my changes fit well into that category.

File: jenkins-scripts/docker/reprepro_updater.bash (hunk 2 above)
Comment (j-rivero) [reply jenkins-scripts/docker/reprepro_updater.bash:21]: no. I use it as a generic failure.

File: jenkins-scripts/docker/reprepro_updater.bash (hunk 3 above)
Comment (j-rivero) [reply jenkins-scripts/docker/reprepro_updater.bash:47]: This comes from the very bad habit of the old server where root was in charge of managing the repository. That configuration did not change while migrating to the new packages machine. Should be changed but there is a good number of places where that needs to happen so a bit of planned work is need to make it happen.

File: jenkins-scripts/dsl/core.dsl [hunk 4]
```diff
@@ -31,7 +31,6 @@ update_vcpkg_snapshot_job.with
     }
 }
 
-
```
Comment (nuclearsandwich) [inline jenkins-scripts/dsl/core.dsl:34]: I don't necessarily mind stray whitespace changes but it is worth asking if this was intentional.

File: jenkins-scripts/docker/reprepro_updater.bash (hunk 1 above)
Comment (j-rivero) [reply jenkins-scripts/docker/reprepro_updater.bash:10]: @nuclearsandwich and I agree on moving my fork to reprepro-updater.

Comment (j-rivero) [reply jenkins-scripts/docker/reprepro_updater.bash]: a83445c949e6e2feb1ab4af797e4c127b6892333

File: jenkins-scripts/dsl/core.dsl (hunk 4 above)
Comment (j-rivero) [reply jenkins-scripts/dsl/core.dsl:34]: it was, yes.

**Review by nuclearsandwich — APPROVED** [review]

**Pull request merged**